queries/cdmq/cdm.js

```diff
@@ -1229,12 +1229,21 @@ getMetricIdsFromTerms = function (url, periId, terms_string) {
             //    queries and aggregate the metric IDs.
             // 3) Simply adjust index.max_result_window to > 10,000, but test this,
             //    as the size is dependent on the Java heap size.
+            //
+  if (periId == null && runId == null) {
+    console.log("ERROR: getMetricIdsFromTerms must have either a period-id or run-id\n");
+    return;
+  }
   if (periId != null) {
     q.query.bool.filter.push(JSON.parse('{"term": {"period.id": "' + periId + '"}}'));
   }
+  if (runId != null) {
+    q.query.bool.filter.push(JSON.parse('{"term": {"run.id": "' + runId + '"}}'));
+  }
   var resp = esRequest(url, "metric_desc/_doc/_search", q);
   var data = JSON.parse(resp.getBody());
-  if (data.hits.total.value >= bigQuerySize) {
+  if (data.hits.total.value >= bigQuerySize || data.hits.hits.length >= bigQuerySize) {
```

Comment:
I think there should be a comment here explaining this...its a ES bug, right?

Reply:
comment added on latest commit